### PR TITLE
ENH: Database 

### LINKED ISSUE #1: Mysql Python connection Python code review
Having code issue in dotenv,  connection issue and code modularization

### PR DESCRIPTION
**Pull request by sumedhkumarprasad**
Please review my python code

### CODE REVIEW

**Comment by sumedhkumarprasad** [comment]
Please review my Python code

File: utils.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+from typing import List, Tuple
+import os
+from os.path import join, dirname
+import logging
+
+from pathlib import Path
+import mysql.connector as mysql
+from dotenv import load_dotenv,dotenv_values
+
+os.chdir(Path("E:/E2E_DTP_Project"))
```
Comment (mitulds) [inline utils.py]: on line 19, you don't need `os.chdir()`, just `Path(..)` should be fine

File: utils.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+"""
+
+from typing import List, Tuple
+import os
+from os.path import join, dirname
+import logging
+
+from pathlib import Path
+import mysql.connector as mysql
+from dotenv import load_dotenv,dotenv_values
```
Comment (mitulds) [inline utils.py]: line 17 - no need to import `dotenv_values`

File: utils.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+logging.basicConfig(filename='myapp.log',
+                    format='%(asctime)s %(message)s', 
+                    level=logging.INFO)
+
+env = Path(".env")
+load_dotenv(dotenv_path=env)
+
+from dotenv import load_dotenv
+load_dotenv()
```
Comment (mitulds) [inline utils.py]: line 29 & 30 repeated code

File: utils.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        user (str): pass user name
+
+    Returns:
+        Tuple[mysql.connection.MySQLConnection, str]: 
+    """
+    try:
+        logging.info('Started')
+        conn = mysql.connect(host=host,
+                            user=user,
+                            password="12345")
```
Comment (mitulds) [inline utils.py]: line 46 - you can place your password in `.env` file. The format should be as follows

`PASSWORD = 12345`

then in your python script you can replace line 46 with `os.getenv("PASSWORD")`

File: utils.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        logging.info("Connected to MySQL successfully!")
+        logging.info('ended')
+    except Exception as e:
+        logging.debug(e)
+    return conn, cur
+
+conn, cur = connect_db(host="localhost", user="root")
+
+import pandas as pd
+import numpy as np
```
Comment (mitulds) [inline utils.py]: all imports should go at the start of the script

File: utils.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        types1.append("INT")   
+        
+types1 
+coltypes = list(zip(sample_data_df.columns, types1)) 
+coltypes  
+coltypes = tuple([' '.join(i) for i in coltypes])
+coltypes 
+coltypes = ', '.join(coltypes)
+coltypes
+'''
```
Comment (mitulds) [inline utils.py]: line 89 - 106 can be removed

**Comment by mitulds** [comment]
You can create a `utils.py` script. All functions that we will ever create will go into this script.

File: utils.py (hunk 1 above)
Comment (sumedhkumarprasad) [reply utils.py]: In actual project, path is required or not

 Line - 19 os.chdir(Path("E:/E2E_DTP_Project") Is it necessary to mention it

File: utils.py (hunk 6 above)
Comment (sumedhkumarprasad) [reply utils.py]: Changes Done

File: utils.py (hunk 5 above)
Comment (sumedhkumarprasad) [reply utils.py]: Changes done

File: utils.py (hunk 4 above)
Comment (sumedhkumarprasad) [reply utils.py]: changes and correction completed

File: utils.py (hunk 3 above)
Comment (sumedhkumarprasad) [reply utils.py]: Removed

File: utils.py (hunk 2 above)
Comment (sumedhkumarprasad) [reply utils.py]: removed dotenv_values. Completed these steps.

File: utils.py (hunk 1 above)
Comment (mitulds) [reply utils.py]: you actually don't even need this line of line

Comment (sumedhkumarprasad) [reply utils.py]: ok I will remove it now from the code

Comment (sumedhkumarprasad) [reply utils.py]: Remove completed

**Pull request closed without merging**